### ISSUE TITLE
bind PayHardwareService failed: service not found

### ISSUE THREAD

**Issue by supsabhi** [SnorSnor9998/Sunmi-Card-Reader#3]
as soon as i press start button from demo app it crashes with this error.. bind PayHardwareService failed: service not found

can you please look into it?

**Comment by SnorSnor9998** [comment]
What device you running & which android verison u at?

**Comment by ghadeer-hammoud** [comment]
Add this permission to the app manifest
<uses-permission android:name="android.permission.QUERY_ALL_PACKAGES" tools:ignore="QueryAllPackagesPermission"/>

**Comment by ghadeer-hammoud** [comment]
Newer Android versions need to explicitly inquire about the needed service. Even add this to query your service package:
`<queries>
        <package android:name="<PACKAGE-NAME>"/>
    </queries>`

Or just add this permission to the app manifest to allow it to inquire:
`<uses-permission android:name="android.permission.QUERY_ALL_PACKAGES" tools:ignore="QueryAllPackagesPermission"/>`

**Comment by SnorSnor9998** [comment]
Yes, all Sunmi Device started Android 11 needed to add `QUERY_ALL_PACKAGES` on manifest.
https://developer.sunmi.com/docs/en-US/xeghjk491/fxzeghjk557#h-4-notes%EF%BC%9A
